src/golang/cmd/server/handler/get_workflow_dag_result.go

```diff
@@ -153,10 +154,43 @@ func (h *GetWorkflowDagResultHandler) Perform(ctx context.Context, interfaceArgs
 		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving artifact results.")
 	}
 
+	contents, err := getArtifactContents(ctx, constructedDag, artifactResults)
+	if err != nil {
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving artifact contents.")
+	}
+
 	return dag.NewResultResponseFromDbObjects(
 		constructedDag,
 		dbWorkflowDagResult,
 		operatorResults,
 		artifactResults,
+		contents,
 	), http.StatusOK, nil
 }
+
+// TODO: We should replace this once we migrate to services based on `artifact` objects
```

Comment:
Can we link a task here? Want to avoid hanging TODO's in our codebase.

Reply:
Can you note in the function comment that we only grab artifact contents for certain small artifact types? Essentially pull up the comment from line 181